Task3/Queue/Queue.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace Queue
+{
+    public class Queue<T> : IQueue<T> where T : struct
+    {
+        private LinkedList<T> _list = new LinkedList<T>();
+
```

Comment:
Since I limit the variable not to be of reference type I think I do not need the check at all. Maybe if I need to perform another validation of the given data I would need it.